src/us/kbase/workspace/database/provenance/Common.java

```diff
@@ -47,38 +52,58 @@ static <T> List<T> processSimpleList(final List<T> list, final String name) {
 			return Common.immutable(list);
 		}
 	}
-	
+
+	/** Check that a PID string is non-null, has at least one non-whitespace character,
+	 * and conforms to the specified regular expression.
+	 * @param putativePid the string to check.
+	 * @param name the name of the string to use in any error messages.
+	 * @return the trimmed PID.
+	 */
+	public static String checkPid(final String putativePid, final String name)
```

Comment:
Changed this to `checkPid` and more generally changed `ResolvablePID` to `PermanentID` as PID is a commonly-used acronym (well, Elisha uses it) but RPID isn't.